tests/integration/query_execute_test.go

```diff
@@ -771,3 +772,82 @@ func TestIssue1785FillDecimalFields(t *testing.T) {
 		require.EqualValues(t, expectedVal, rd.DecimalVal)
 	})
 }
+
+// https://github.com/ydb-platform/ydb-go-sdk/issues/1872
+func TestIssue1872QueryWarning(t *testing.T) {
+	ctx, cancel := context.WithCancel(xtest.Context(t))
+	defer cancel()
+	db, err := ydb.Open(ctx,
+		os.Getenv("YDB_CONNECTION_STRING"),
+		ydb.WithAccessTokenCredentials(os.Getenv("YDB_ACCESS_TOKEN_CREDENTIALS")),
+		ydb.WithTraceQuery(
+			log.Query(
+				log.Default(os.Stdout,
+					log.WithLogQuery(),
+					log.WithColoring(),
+					log.WithMinLevel(log.INFO),
+				),
+				trace.QueryEvents,
+			),
+		),
+	)
+	require.NoError(t, err)
+	_ = db.Query().Exec(ctx,
+		`drop table TestIssue1872QueryWarning;`,
+	)
+	err = db.Query().Exec(ctx,
+		`create table TestIssue1872QueryWarning 
+		(Id uint64, Amount decimal(22,9) , primary key(Id));`,
+		query.WithParameters(
+			ydb.ParamsBuilder().
+				Param("$p1").Text("test1").
+				Build(),
+		),
+	)
+	require.NoError(t, err)
+
+	t.Run("Query", func(t *testing.T) {
+		var issueList []*Ydb_Issue.IssueMessage
+		q := db.Query()
+		result, err := q.Query(ctx, `
+        insert into TestIssue1872QueryWarning (Id, Amount) values (-3, Decimal("3.01",22,9));
+		insert into TestIssue1872QueryWarning (Id, Amount) values (-5, Decimal("5.01",22,9));
+        `,
+			query.WithParameters(
+				ydb.ParamsBuilder().
+					Param("$p1").Text("test").
+					Build(),
+			),
+			query.WithSyntax(query.SyntaxYQL),
+			query.WithIdempotent(),
+			query.WithIssuesHandler(func(issues []*Ydb_Issue.IssueMessage) {
+				issueList = issues
```

Comment:
You can add asserts inside callback:

```suggestion
                 require.Len(t, issueList, 1)
				require.Equal(t, "Failed to ...", issueList[0].Issues[0].Issues[0].Message)
```

and other asserts

Reply:
ok